From: R. Bernstein <rocky@gnu.org>
Date: Thu, 29 Oct 2009 14:16:55 +0000 (-0400)
Subject: Change tests for S_ISLNK and S_ISSOCK macros from AC_COMPILE_IFELSE to
X-Git-Url: http://git.savannah.gnu.org/gitweb/?p=libcdio.git;a=commitdiff_plain;h=9bec7fc20431c3a6527ea419eaceebe49737d9dc;hp=423dc300f64a6364796b7d0ca4ddf9df2d7dd7a5

Change tests for S_ISLNK and S_ISSOCK macros from AC_COMPILE_IFELSE to
AC_LINK_IFELSE. On Mingw 4.4.0 these macros succeed when they should
fail. Patch by carlo.bramix to libcdio-devel
---

diff --git a/configure.ac b/configure.ac
index be1c901..0414085 100644
--- a/configure.ac
+++ b/configure.ac
@@ -217,7 +217,7 @@ AC_MSG_RESULT($ISOC99_PRAGMA)
 ## Check for S_ISSOCK() and S_ISLNK() macros
 ##
 AC_MSG_CHECKING(for S_ISLNK() macro)
-AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
+AC_LINK_IFELSE([AC_LANG_PROGRAM([
 #ifdef HAVE_SYS_STAT_H
 # include <sys/stat.h>
 #endif
@@ -226,7 +226,7 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
 [ AC_MSG_RESULT(no) ])
 
 AC_MSG_CHECKING([for S_ISSOCK() macro])
-AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
+AC_LINK_IFELSE([AC_LANG_PROGRAM([
 #ifdef HAVE_SYS_STAT_H
 # include <sys/stat.h>
 #endif
